components/gfps_service/include/gfps.hpp

```diff
@@ -43,9 +43,34 @@ namespace gfps {
 /// @param length The length of the data
 typedef std::function<bool(nearby_fp_Characteristic, const uint8_t *, size_t)> notify_callback_t;
 
+/// Callback for assigning a passkey to the pairing peer
+/// @param passkey The passkey value to be injected
+typedef std::function<void(uint32_t)> set_passkey_callback_t;
+
+/// Callback invoked when an Account Key is written by a remote device
+/// @param peer_addr BLE address of the remote device
+/// @param key Pointer to the 16-byte Account Key data
+typedef std::function<void(uint64_t, const uint8_t *key)> account_key_write_callback_t;
+
+/// Callback triggered when the GFPS layer finishes constructing a Non-Discoverable Advertisement payload
+/// @param adv_data Pointer to raw advertisement payload bytes
+/// @param len Length of the advertisement data
+typedef std::function<void(const uint8_t *adv_data, size_t len)> nda_ready_callback_t;
```

Comment:
For these two functions, can we change the use of `const uint8_t*` and `const uint8_t*, size_t` to be `const std::span<uint8_t>&`. Given that the first function's argument is known size of 16 bytes, we can use

```c++
const std::span<uint8_t, 16>&
```

While for the second, we'd still use

```c++
const std::span<uint8_t>&
```

This provides a lightweight accessor which enables the application code to use range-based methods and other accessors, while also including OOB checking and other type/memory safety helpers.

Then in the event handler which calls these callbacks you can simply call:

```c++
g_account_key_write_callback_(peer_addr, std::span<uint8_t, 16>(key, 16));
```

and

```c++
g_nda_ready_callback_(std::span<uint8_t>(adv_data, len));
```